### ISSUE TITLE
[Feature]

### ISSUE THREAD

**Issue by jefree-cat** [oam-dev/kubevela#3527]
**Is your feature request related to a problem? Please describe.**
Use Case: frontend workload + related trait
In micro-service scenario, typical  application toplogy is 3 components architecture(frontend + backend(webservice) + database). for application developer,  frontend(etc: react) workload has some unique traits distinguish from webservice(spring-boot). For example,
1. Observability is unique: frontend logging, frontend metrics(in browser)
2. Configuration Management: Nginx configs, etc.
3. Normally depends on backend svc(webservice). It means workflow(deploy/cleanup) sequence has specific points.

**Describe the solution you'd like**
In short, frontend is typical workload type.
Should provide frontend workload with related traits although it has some common parts with webservice workload.

**Additional context**
Just a example from digitalOcean
https://www.digitalocean.com/community/tutorials/how-to-deploy-a-react-application-to-digitalocean-app-platform

**Comment by jefree-cat** [comment]
@wonderflow, please review it. thx.

**Comment by barnettZQG** [comment]
@jefree-cat As you said, It differs from backend stateless services only in traits. so, maybe you could define some trait definitions, but you can also use the webservice component definition.

**Comment by wonderflow** [comment]
Let's make the frontend workload as addon.

**Comment by jefree-cat** [comment]
> Let's make the frontend workload as addon.
will handle it.

**Comment by jefree-cat** [comment]
will handle it.

**Comment by briankane** [comment]
This issue has been inactive for over 2 years and is scheduled for closure as part of our backlog maintenance.

  **If this issue is still relevant:**
  - Please comment below to keep it open
  - Provide updated context if the original description is outdated

  **Timeline:** If there's no activity by **Friday, October 24th**, this issue will be automatically closed.

  **Note:** Issues can always be reopened if they become relevant again. We encourage the community to reopen and reprioritize any incorrectly 
  closed issues.

**Comment by briankane** [comment]
This issue has been closed following the stale issue notice. 

If this issue is still relevant and should remain open, please:
1. Reopen the issue
2. Remove the 'stale' label
3. Provide an update on the current status or importance

Thank you for your understanding and contribution to KubeVela!
